pkg/controller/registry/configmap_loader.go

```diff
@@ -21,23 +21,22 @@ const (
 
 // ConfigMapCatalogResourceLoader loads a ConfigMap of resources into the in-memory catalog
 type ConfigMapCatalogResourceLoader struct {
-	Catalog   *InMem
```

Comment:
Why the change in this interface?

Reply:
This is fixing the memory leak, since the catalog returned was a field of the loader, and the loader holds a reference to part of the operator (operator client), every time we sync the catalog we were loading into memory and retaining it indefinitely. 

We might still be leaking the loader itself now that you mention it...but at least that's a tiny amount instead of an entire in memory catalog